packages/transaction-controller/src/TransactionController.test.ts

```diff
@@ -127,6 +134,20 @@ function mockFetchWithDynamicResponse(dataForUrl: any) {
     );
 }
 
+/**
+ * Builds a mock block tracker with a canned block number that can be used in
+ * tests.
+ *
+ * @param latestBlockNumber - The block number that the block tracker should
+ * always return.
+ * @returns The mocked block tracker.
+ */
+function buildMockBlockTracker(latestBlockNumber: string): BlockTrackerProxy {
```

Comment:
BlockTracker takes a SafeEventEmitterProvider, but we aren't using that for our fake provider yet, so I can't use the same approach I was using in #1116. This approach seems to work, however. This did mean that I had to pull out FakeBlockTracker instead of FakeProvider to `tests/`.

Also for some reason I couldn't use `jest.spyOn(fakeBlockTracker, 'getLatestBlock').mockResolvedValue(latestBlockNumber)` here, I had to add a new method to the FakeBlockTracker class to allow for us to specify a block number. Not sure if this is the proxy getting in the way or not.